src/Elmah/XmlFileErrorLog.cs

```diff
@@ -158,10 +173,36 @@ public override string Log(Error error)
                 writer.Flush();
             }
 
+            DeleteOldFiles();
             return errorId;
         }
 
         /// <summary>
+        /// Deletes Old files according the <code>size</code> parameter
+        /// </summary>
+        private void DeleteOldFiles()
+        {
+            try
+            {
+                if (!Directory.Exists(LogPath))
+                    return;
+
+                DirectoryInfo dir = new DirectoryInfo(LogPath);
+                FileInfo[] logFiles = dir.GetFiles("error*.xml");
+
+                foreach (var file in logFiles.OrderByDescending(f => f.CreationTime).Skip(_fileListSize))
+                {
+                    file.Delete();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException(string.Format("Error during cleaning old log files for the XML file-based error log. Exceprion: {0}", ex.StackTrace));
```

Comment:
Use `Exception` instead of `ApplicationException` because this is not an application-related error. Also, wrap the inner exception (`ex`) instead of dumping it's stack trace into a _new_ & unrelated exception.